## Objectivos

Luego de completar este lab seras capas de:

*   Adquirir información de varias formas.
*   Obtener conocimientos a partir de datos con la librería Pandas.

<h2>Tabla de Contenidos</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ol>
    <li><a href="https://#data_acquisition">Obtención de la información</a>
    <li><a href="https://#basic_insight"> Conocimiento básico de un Dataset</a></li>
</ol>

</div>
<hr>


<h1 id="data_acquisition">Adquisición de la Información</h1>
<p>
Hay varios formatos para un dataset: .csv, .json, .xlsx  etc. El dataset puede ser ordenado en diferentes lugares, en tu maquina local o a veces en linea.<br>

En esta sección, aprenderás a cargar un dataset a un Jupyter Notebook.<br>

En nuestro caso, El Dataset Automobile es una fuente de recursos online, y esta en formato CSV (`C`omma `S`eparated `V`alue/Valores Separados por Coma). Vamos a usar este dataset como ejemplo para practicar la lectura de datos.

<ul>
    <li>Fuente de la información: <a href="https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01">https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data</a></li>
    <li>Tipo de información: csv</li>
</ul>
La librearía pandas es una herramienta util que te permite leer varios dataset y cargarlos a un dataframe el ambiente virtual ya tiene instalado <b>Pandas</b> asi que solo debes elegir el kernel e importar la librearía.
</p>


In [1]:
# Importando las librerías.
import pandas as pd
import numpy as np

<h2>Leyendo información</h2>
<p>
Usamos la función <code>pandas.read_csv()</code> para leer un archivo csv. En los brackets, ponemos la dirección del archivo envuelto en comillas para que asi pandas lo cargue a un dataframe. La dirección del archivo puede una URL o una dirección local de tus archivos.<br>

Ya que la información no incluye encabezados, podemos agregar un argumento <code>headers = None</code> dentro del método <code>read_csv()</code> para que de esta manera <b>Pandas</b> no ajuste de forma automática la primera fila como un encabezado.<br>

También puedes asignar al dataset cualquier variable que hayas creadoYou can also assign the dataset to any variable you create.
</p>

In [2]:
path = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DA0101EN-SkillsNetwork/labs/Data%20files/auto.csv"

Necesitas descargar el dataset; si estas corriendo el lab de manera local.

In [3]:
# Descomenta esta linea si descargaste el archivo de manera local.
#path ="auto.csv"

In [4]:
# Leemos el archivo en linea de la URL proporcionada arriba, y lo asignamos a una variable llamada "df"

df = pd.read_csv(path, header=None)

Luego de leer el dataset, podemos usar el método `dataframe.head(n)` para revisar la `n` cantidad de filas en el dataframe, donde `n` es un integer. Contrario a `dataframe.head(n)`, es `dataframe.tail(n)` el cual te mostrara la `n` cantidad de filas del fondo del dataframe.

In [5]:
# Muestra las primeras cinco filas usando el método df.head()
df.head(5)

0    1            2    3    4     5            6    7      8     9   ...  \
0   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
1   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
2   1    ?  alfa-romero  gas  std   two    hatchback  rwd  front  94.5  ...   
3   2  164         audi  gas  std  four        sedan  fwd  front  99.8  ...   
4   2  164         audi  gas  std  four        sedan  4wd  front  99.4  ...   

    16    17    18    19    20   21    22  23  24     25  
0  130  mpfi  3.47  2.68   9.0  111  5000  21  27  13495  
1  130  mpfi  3.47  2.68   9.0  111  5000  21  27  16500  
2  152  mpfi  2.68  3.47   9.0  154  5000  19  26  16500  
3  109  mpfi  3.19  3.40  10.0  102  5500  24  30  13950  
4  136  mpfi  3.19  3.40   8.0  115  5500  18  22  17450  

[5 rows x 26 columns]

**Pregunta #1** Revisa las ultimas diez filas del dataframe "df"

In [6]:
# Escribe tu código abajo
print("Print de las ultimas diez lineas del dataframe")
df.tail(10)

Print de las ultimas diez lineas del dataframe


0    1      2       3      4     5      6    7      8      9   ...   16  \
195  -1   74  volvo     gas    std  four  wagon  rwd  front  104.3  ...  141   
196  -2  103  volvo     gas    std  four  sedan  rwd  front  104.3  ...  141   
197  -1   74  volvo     gas    std  four  wagon  rwd  front  104.3  ...  141   
198  -2  103  volvo     gas  turbo  four  sedan  rwd  front  104.3  ...  130   
199  -1   74  volvo     gas  turbo  four  wagon  rwd  front  104.3  ...  130   
200  -1   95  volvo     gas    std  four  sedan  rwd  front  109.1  ...  141   
201  -1   95  volvo     gas  turbo  four  sedan  rwd  front  109.1  ...  141   
202  -1   95  volvo     gas    std  four  sedan  rwd  front  109.1  ...  173   
203  -1   95  volvo  diesel  turbo  four  sedan  rwd  front  109.1  ...  145   
204  -1   95  volvo     gas  turbo  four  sedan  rwd  front  109.1  ...  141   

       17    18    19    20   21    22  23  24     25  
195  mpfi  3.78  3.15   9.5  114  5400  23  28  13415  
196  mpfi  3.78  3.15   9.5  114  5400  24  28  15985  
197  mpfi  3.78  3.15   9.5  114  5400  24  28  16515  
198  mpfi  3.62  3.15   7.5  162  5100  17  22  18420  
199  mpfi  3.62  3.15   7.5  162  5100  17  22  18950  
200  mpfi  3.78  3.15   9.5  114  5400  23  28  16845  
201  mpfi  3.78  3.15   8.7  160  5300  19  25  19045  
202  mpfi  3.58  2.87   8.8  134  5500  18  23  21485  
203   idi  3.01  3.40  23.0  106  4800  26  27  22470  
204  mpfi  3.78  3.15   9.5  114  5400  19  25  22625  

[10 rows x 26 columns]

<details><summary>Click here for the solution</summary>

```python
print("The last 10 rows of the dataframe\n")
df.tail(10)
```

<h3>Agregando encabezados (Headers)</h3>

Revisa nuestro dataset. `Pandas` automáticamente prepara los encabezados con un integer comenzando desde el 0.

Para una mejor descripción de nuestra información de nuestra información, podemos introducir encabezados. esta información se encuentra disponible en:  <a href="https://archive.ics.uci.edu/ml/datasets/Automobile?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01">https://archive.ics.uci.edu/ml/datasets/Automobile</a>.

Por lo tanto, debemos agregar nuestros encabezados de forma manual.

**Primero**, creamos una lista "headers" que incluye todos los nombres de las columnas en orden.
Luego, usamos `dataframe.columns = headers` para reemplazar los encabezados con la lista que creamos.


In [7]:
# Creamos la lista headers
headers = ["symboling","normalized-losses","make","fuel-type","aspiration", "num-of-doors","body-style",
         "drive-wheels","engine-location","wheel-base", "length","width","height","curb-weight","engine-type",
         "num-of-cylinders", "engine-size","fuel-system","bore","stroke","compression-ratio","horsepower",
         "peak-rpm","city-mpg","highway-mpg","price"]
print("headers\n", headers)

headers
 ['symboling', 'normalized-losses', 'make', 'fuel-type', 'aspiration', 'num-of-doors', 'body-style', 'drive-wheels', 'engine-location', 'wheel-base', 'length', 'width', 'height', 'curb-weight', 'engine-type', 'num-of-cylinders', 'engine-size', 'fuel-system', 'bore', 'stroke', 'compression-ratio', 'horsepower', 'peak-rpm', 'city-mpg', 'highway-mpg', 'price']


Reemplazamos los encabezados y volvemos a revisar nuestro dataframe.

In [8]:
df.columns = headers
df.head(10)

symboling normalized-losses         make fuel-type aspiration num-of-doors  \
0          3                 ?  alfa-romero       gas        std          two   
1          3                 ?  alfa-romero       gas        std          two   
2          1                 ?  alfa-romero       gas        std          two   
3          2               164         audi       gas        std         four   
4          2               164         audi       gas        std         four   
5          2                 ?         audi       gas        std          two   
6          1               158         audi       gas        std         four   
7          1                 ?         audi       gas        std         four   
8          1               158         audi       gas      turbo         four   
9          0                 ?         audi       gas      turbo          two   

    body-style drive-wheels engine-location  wheel-base  ...  engine-size  \
0  convertible          rwd           front        88.6  ...          130   
1  convertible          rwd           front        88.6  ...          130   
2    hatchback          rwd           front        94.5  ...          152   
3        sedan          fwd           front        99.8  ...          109   
4        sedan          4wd           front        99.4  ...          136   
5        sedan          fwd           front        99.8  ...          136   
6        sedan          fwd           front       105.8  ...          136   
7        wagon          fwd           front       105.8  ...          136   
8        sedan          fwd           front       105.8  ...          131   
9    hatchback          4wd           front        99.5  ...          131   

   fuel-system  bore  stroke compression-ratio horsepower  peak-rpm city-mpg  \
0         mpfi  3.47    2.68               9.0        111      5000       21   
1         mpfi  3.47    2.68               9.0        111      5000       21   
2         mpfi  2.68    3.47               9.0        154      5000       19   
3         mpfi  3.19    3.40              10.0        102      5500       24   
4         mpfi  3.19    3.40               8.0        115      5500       18   
5         mpfi  3.19    3.40               8.5        110      5500       19   
6         mpfi  3.19    3.40               8.5        110      5500       19   
7         mpfi  3.19    3.40               8.5        110      5500       19   
8         mpfi  3.13    3.40               8.3        140      5500       17   
9         mpfi  3.13    3.40               7.0        160      5500       16   

  highway-mpg  price  
0          27  13495  
1          27  16500  
2          26  16500  
3          30  13950  
4          22  17450  
5          25  15250  
6          25  17710  
7          25  18920  
8          20  23875  
9          22      ?  

[10 rows x 26 columns]

Necesitamos reemplazar el símbolo "?" con NaN asi que usamos el método dropna() para remover los valores faltantes.

In [9]:
# Reemplazamos los valore "?" por NaN
df1 = df.replace("?",np.NaN)

# Revisamos nuestro dataframe para ver que los valore "?" hayan sido cambiados por NaN.
df1.head()

symboling normalized-losses         make fuel-type aspiration num-of-doors  \
0          3               NaN  alfa-romero       gas        std          two   
1          3               NaN  alfa-romero       gas        std          two   
2          1               NaN  alfa-romero       gas        std          two   
3          2               164         audi       gas        std         four   
4          2               164         audi       gas        std         four   

    body-style drive-wheels engine-location  wheel-base  ...  engine-size  \
0  convertible          rwd           front        88.6  ...          130   
1  convertible          rwd           front        88.6  ...          130   
2    hatchback          rwd           front        94.5  ...          152   
3        sedan          fwd           front        99.8  ...          109   
4        sedan          4wd           front        99.4  ...          136   

   fuel-system  bore  stroke compression-ratio horsepower  peak-rpm city-mpg  \
0         mpfi  3.47    2.68               9.0        111      5000       21   
1         mpfi  3.47    2.68               9.0        111      5000       21   
2         mpfi  2.68    3.47               9.0        154      5000       19   
3         mpfi  3.19    3.40              10.0        102      5500       24   
4         mpfi  3.19    3.40               8.0        115      5500       18   

  highway-mpg  price  
0          27  13495  
1          27  16500  
2          26  16500  
3          30  13950  
4          22  17450  

[5 rows x 26 columns]

Podemos votar los valores perdidos a lo largo de la columna "price" de la siguiente manera:

In [10]:
df = df1.dropna(subset=["price"], axis=0)

# Revisamos las primeras 20 filas para asegurarnos de que no existen valores NaN en la columna "price"
df.head(20)

symboling normalized-losses         make fuel-type aspiration  \
0           3               NaN  alfa-romero       gas        std   
1           3               NaN  alfa-romero       gas        std   
2           1               NaN  alfa-romero       gas        std   
3           2               164         audi       gas        std   
4           2               164         audi       gas        std   
5           2               NaN         audi       gas        std   
6           1               158         audi       gas        std   
7           1               NaN         audi       gas        std   
8           1               158         audi       gas      turbo   
10          2               192          bmw       gas        std   
11          0               192          bmw       gas        std   
12          0               188          bmw       gas        std   
13          0               188          bmw       gas        std   
14          1               NaN          bmw       gas        std   
15          0               NaN          bmw       gas        std   
16          0               NaN          bmw       gas        std   
17          0               NaN          bmw       gas        std   
18          2               121    chevrolet       gas        std   
19          1                98    chevrolet       gas        std   
20          0                81    chevrolet       gas        std   

   num-of-doors   body-style drive-wheels engine-location  wheel-base  ...  \
0           two  convertible          rwd           front        88.6  ...   
1           two  convertible          rwd           front        88.6  ...   
2           two    hatchback          rwd           front        94.5  ...   
3          four        sedan          fwd           front        99.8  ...   
4          four        sedan          4wd           front        99.4  ...   
5           two        sedan          fwd           front        99.8  ...   
6          four        sedan          fwd           front       105.8  ...   
7          four        wagon          fwd           front       105.8  ...   
8          four        sedan          fwd           front       105.8  ...   
10          two        sedan          rwd           front       101.2  ...   
11         four        sedan          rwd           front       101.2  ...   
12          two        sedan          rwd           front       101.2  ...   
13         four        sedan          rwd           front       101.2  ...   
14         four        sedan          rwd           front       103.5  ...   
15         four        sedan          rwd           front       103.5  ...   
16          two        sedan          rwd           front       103.5  ...   
17         four        sedan          rwd           front       110.0  ...   
18          two    hatchback          fwd           front        88.4  ...   
19          two    hatchback          fwd           front        94.5  ...   
20         four        sedan          fwd           front        94.5  ...   

    engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
0           130         mpfi  3.47    2.68               9.0        111   
1           130         mpfi  3.47    2.68               9.0        111   
2           152         mpfi  2.68    3.47               9.0        154   
3           109         mpfi  3.19    3.40              10.0        102   
4           136         mpfi  3.19    3.40               8.0        115   
5           136         mpfi  3.19    3.40               8.5        110   
6           136         mpfi  3.19    3.40               8.5        110   
7           136         mpfi  3.19    3.40               8.5        110   
8           131         mpfi  3.13    3.40               8.3        140   
10          108         mpfi  3.50    2.80               8.8        101   
11          108         mpfi  3.50    2.80               8.8        101   
12          164

Ahora, hemos leído exitosamente un dataset en bruto y agregado correctamente los headers al dataframe.

**Pregunta 2** 
Encuentra el nombre de las columnas del dataframe.

In [12]:
# Escribe tu respuesta acá
df.columns

Index(['symboling', 'normalized-losses', 'make', 'fuel-type', 'aspiration',
       'num-of-doors', 'body-style', 'drive-wheels', 'engine-location',
       'wheel-base', 'length', 'width', 'height', 'curb-weight', 'engine-type',
       'num-of-cylinders', 'engine-size', 'fuel-system', 'bore', 'stroke',
       'compression-ratio', 'horsepower', 'peak-rpm', 'city-mpg',
       'highway-mpg', 'price'],
      dtype='object')

<details><summary>Click here for the solution</summary>

```python
print(df.columns)
```

</details>

### Guardando un Dataset.

Correspondientemente, Pandas nos permite guardar información en formato csv. Al usar el método `dataframe.to_csv()`, puedes agregar la ruta y el nombre del archivo entre comillas dentro de los brackets.

Por ejemplo, si fueras a guardar el dataframe **df** como `automobile.csv` en tu maquina local, puedes usar la sintaxis de abajo, donde `index = False` significa que la fila de los nombres no sera escrita.

df.to_csv("automobile.csv",index = False)

También podemos leer o guardar otros formatos de archivo. Podemos usar funciones similares como **`pd.read_csv()`** y **`df.to_csv()`** para otros formatos. Las funciones están listadas en la siguiente tabla:

### Leer/Guardar información en otros Formatos.
| Tipo de formato |        Leer       |         Guardar |
| ------------    | :---------------: | --------------: |
| csv             |  `pd.read_csv()`  |   `df.to_csv()` |
| json            |  `pd.read_json()` |  `df.to_json()` |
| excel           | `pd.read_excel()` | `df.to_excel()` |
| hdf             |  `pd.read_hdf()`  |   `df.to_hdf()` |
| sql             |  `pd.read_sql()`  |   `df.to_sql()` |
| ...             |        ...        |             ... |

<h1 id="basic_insight">Conocimiento básico del Dataset</h1>

Luego de cargar la información al dataframe de *Pandas*, es tiempo de que exploremos el dataset.

Hay varias formas de obtener conocimiento esencial de la información para ayudarnos a entender mejor nuestro dataset.

## Tipos de Información

La información tiene una amplia variedad de tipos.

Los principales tipos almacenados en los dataframe de *Pandas* son **object**, **float**, **int**, **bool** y **datetime64**. Par poder aprender mejor acerca de cada atributo, Siempre es bueno saber el tipo de información de cada columna. En *Pandas*:

In [13]:
# Para saber que tipo de información tiene cada columna usamos la propiedad dtypes
df.dtypes

symboling              int64
normalized-losses     object
make                  object
fuel-type             object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                  object
stroke                object
compression-ratio    float64
horsepower            object
peak-rpm              object
city-mpg               int64
highway-mpg            int64
price                 object
dtype: object

Una serie con el tipo de información de cada columna nos ha sido devuelta.

Como se muestra arriba, es fácil de ver que los tipos de información para "symboling" y "curb-weight" son **int64**, "normalized-losses" es **object**, y "wheel-base" es **float64**, etc.

Estos tipos de información pueden ser modificados; aprenderemos como hacer esto en módulos futuros.

## Describe

Si quisiéramos tener un sumario estadístico de cada columna e.j. count, valor medio de la columna, desviación estándar de la columna , etc., usamos el método describe:

**dataframe.describe()**

Este método nos proporciona un sumario estadístico variado, excluyendo los valores **NaN** (Not a Number).

In [14]:
df.describe()

symboling  wheel-base      length       width      height  \
count  201.000000  201.000000  201.000000  201.000000  201.000000   
mean     0.840796   98.797015  174.200995   65.889055   53.766667   
std      1.254802    6.066366   12.322175    2.101471    2.447822   
min     -2.000000   86.600000  141.100000   60.300000   47.800000   
25%      0.000000   94.500000  166.800000   64.100000   52.000000   
50%      1.000000   97.000000  173.200000   65.500000   54.100000   
75%      2.000000  102.400000  183.500000   66.600000   55.500000   
max      3.000000  120.900000  208.100000   72.000000   59.800000   

       curb-weight  engine-size  compression-ratio    city-mpg  highway-mpg  
count   201.000000   201.000000         201.000000  201.000000   201.000000  
mean   2555.666667   126.875622          10.164279   25.179104    30.686567  
std     517.296727    41.546834           4.004965    6.423220     6.815150  
min    1488.000000    61.000000           7.000000   13.000000    16.000000  
25%    2169.000000    98.000000           8.600000   19.000000    25.000000  
50%    2414.000000   120.000000           9.000000   24.000000    30.000000  
75%    2926.000000   141.000000           9.400000   30.000000    34.000000  
max    4066.000000   326.000000          23.000000   49.000000    54.000000

Esto muestra un sumario estadístico de todas las columnas de tipo numéricos 

Por ejemplo "symboling" posee 205 conteos, el valor promedio de esta columna es de 0.83, la desviación estándar es de 1.25, el valor mínimo es de -2the mean value of this column is 0.83, the standard deviation is 1.25, the minimum value is -2, El percentil 25 es 0, el percentil 50 es 1, el percentil 75 es 2 y el valor máximo es 3.

Sin embargo, que pasa si quisiéramos revisar todas las columnas incluidas aquellas de tipo object?

Puedes agregar el argumento `include = "all"` dentro del bracket. Vamos a intentarlo nuevamente.

In [15]:
# Describimos todas las columnas en "df"

df.describe(include="all")

symboling normalized-losses    make fuel-type aspiration  \
count   201.000000               164     201       201        201   
unique         NaN                51      22         2          2   
top            NaN               161  toyota       gas        std   
freq           NaN                11      32       181        165   
mean      0.840796               NaN     NaN       NaN        NaN   
std       1.254802               NaN     NaN       NaN        NaN   
min      -2.000000               NaN     NaN       NaN        NaN   
25%       0.000000               NaN     NaN       NaN        NaN   
50%       1.000000               NaN     NaN       NaN        NaN   
75%       2.000000               NaN     NaN       NaN        NaN   
max       3.000000               NaN     NaN       NaN        NaN   

       num-of-doors body-style drive-wheels engine-location  wheel-base  ...  \
count           199        201          201             201  201.000000  ...   
unique            2          5            3               2         NaN  ...   
top            four      sedan          fwd           front         NaN  ...   
freq            113         94          118             198         NaN  ...   
mean            NaN        NaN          NaN             NaN   98.797015  ...   
std             NaN        NaN          NaN             NaN    6.066366  ...   
min             NaN        NaN          NaN             NaN   86.600000  ...   
25%             NaN        NaN          NaN             NaN   94.500000  ...   
50%             NaN        NaN          NaN             NaN   97.000000  ...   
75%             NaN        NaN          NaN             NaN  102.400000  ...   
max             NaN        NaN          NaN             NaN  120.900000  ...   

        engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
count    201.000000          201   197     197        201.000000        199   
unique          NaN            8    38      36               NaN         58   
top             NaN         mpfi  3.62    3.40               NaN         68   
freq            NaN           92    23      19               NaN         19   
mean     126.875622          NaN   NaN     NaN         10.164279        NaN   
std       41.546834          NaN   NaN     NaN          4.004965        NaN   
min       61.000000          NaN   NaN     NaN          7.000000        NaN   
25%       98.000000          NaN   NaN     NaN          8.600000        NaN   
50%      120.000000          NaN   NaN     NaN          9.000000        NaN   
75%      141.000000          NaN   NaN     NaN          9.400000        NaN   
max      326.000000          NaN   NaN     NaN         23.000000        NaN   

        peak-rpm    city-mpg highway-mpg price  
count        199  201.000000  201.000000   201  
unique        22         NaN         NaN   186  
top         5500         NaN         NaN  8921  
freq          36         NaN         NaN     2  
mean         NaN   25.179104   30.686567   NaN  
std          NaN    6.423220    6.815150   NaN  
min          NaN   13.000000   16.000000   NaN  
25%          NaN   19.000000   25.000000   NaN  
50%          NaN   24.000000   30.000000   NaN  
75%          NaN   30.000000   34.000000   NaN  
max          NaN   49.000000   54.000000   NaN  

[11 rows x 26 columns]

Ahora nos entrega el sumario estadístico de todas las columnas, incluyendo los atributos tipo-object

Ahora podemos ver cuantos valores únicos hay, cual de ellos es el valor mas alto y la frecuencia del valor mas alto en las columnas tipo-object

Algunos valores en la tabla de arriba son mostrados como "NaN". Esto es por que esos números no están disponibles independiente de la columna en particular.

### Pregunta #3: 

Puedes seleccionar las columnas del dataframe al indicar el nombre de cada columna. Por ejemplo puedes seleccionar 3 columnas de la siguiente manera:
>`dataframe[[' column 1 ',column 2', 'column 3']]`


Donde "column" es el nombre de la columna, puedes aplicar el método `".describe()"` para obtener la estadística de esas columnas de la siguiente manera: 
>`dataframe[[' column 1 ',column 2', 'column 3'] ].describe()`


Aplica el método `".describe()"` a las columnas 'length' y 'compression-ratio'.

In [17]:
# Escribe tu código mas abajo

df[["length","compression-ratio"]].describe()

length  compression-ratio
count  201.000000         201.000000
mean   174.200995          10.164279
std     12.322175           4.004965
min    141.100000           7.000000
25%    166.800000           8.600000
50%    173.200000           9.000000
75%    183.500000           9.400000
max    208.100000          23.000000

<details><summary>Click here for the solution</summary>

```python
df[['length', 'compression-ratio']].describe()
```
</details>

## Info
Otro método que puedes usar para revisar tu dataset es:
> dataframe.info()

Provee de un sumario conciso de tu Dataframe

Este método imprime la información acerca del Dataframe incluyendo el indice, dtype y columnas,valores no-nulos (non-null) y uso de la memoria.


In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 201 entries, 0 to 204
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   symboling          201 non-null    int64  
 1   normalized-losses  164 non-null    object 
 2   make               201 non-null    object 
 3   fuel-type          201 non-null    object 
 4   aspiration         201 non-null    object 
 5   num-of-doors       199 non-null    object 
 6   body-style         201 non-null    object 
 7   drive-wheels       201 non-null    object 
 8   engine-location    201 non-null    object 
 9   wheel-base         201 non-null    float64
 10  length             201 non-null    float64
 11  width              201 non-null    float64
 12  height             201 non-null    float64
 13  curb-weight        201 non-null    int64  
 14  engine-type        201 non-null    object 
 15  num-of-cylinders   201 non-null    object 
 16  engine-size        201 non-null